packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -387,36 +389,33 @@ export class RunningSummarizer implements IDisposable {
         }
     }
 
-    private trySummarize(reason: string): { broadcastP: Promise<void> } {
-        if (this.summarizing === true) {
+    private trySummarize(reason: string): void {
+        if (this.summarizing !== undefined) {
             // We can't summarize if we are already
             this.tryWhileSummarizing = true;
-            return { broadcastP: Promise.resolve() };
+            return;
         }
 
         // GenerateSummary could take some time
         // mark that we are currently summarizing to prevent concurrent summarizing
-        this.summarizing = true;
-        const broadcastDeferred = new Deferred<void>();
+        this.summarizing = new Deferred();
 
         (async () => {
-            const result = await this.summarize(reason, false, broadcastDeferred);
+            const result = await this.summarize(reason, false);
             if (result !== true) {
                 // On nack or error, try again in safe mode
-                await this.summarize(reason, true, broadcastDeferred);
+                await this.summarize(reason, true);
             }
         })().finally(() => {
-            this.summarizing = false;
-            broadcastDeferred.resolve();
+            this.summarizing?.resolve();
+            this.summarizing = undefined;
             if (this.tryWhileSummarizing) {
                 this.tryWhileSummarizing = false;
                 this.heuristics.run();
             }
         }).catch((error) => {
             this.logger.sendErrorEvent({ eventName: "UnexpectedSummarizeError" }, error);
```

Comment:
I think we want to reject the promise here, no?

Reply:
Why reject?

Reply:
Oh, it's fire-and-forget? Sorry, missed that, then yes, we do not care.